EIPS/eip-5875.md

```diff
@@ -0,0 +1,67 @@
+---
+eip: 5875
+title: Opcode for TX Number
+description: Add an opcode to retrieve TX number
+author: Zainan Victor Zhou (@xinbenlv)
+discussions-to: https://ethereum-magicians.org/t/eip-5875-opcode-for-tx-number-in-a-block/11612
+status: Draft
+type: Standards Track
+category: Core
+created: 2022-11-04
+---
+
+## Abstract
+
+This EIP adds an opcode instruction, `OPCODE_TBD (TXNUM)` to retrieve the transaction number inside of a block.
+
+## Motivation
+
+A transaction is an atomic point of time for a world state.
+
+Currently, we have access to the block number through `BLOCKNUMBER (0x43)` but there is no way to access a transaction number within a smart contract.
+
+This EIP proposes an opcode to fetch the transaction number, which can be used to:
+
+* Identify the exact ordering of transactions within a block
+* Uniquely and unambiguously identify the world state (better than `BLOCKNUMBER`)
+* Allow simple arithmetic on the transaction number (better than an opcode to get the transaction hash)
+* On-chain and off-chain indexing and snapshotting
+
+An example use-case is snapshot weighted voting, in which a snapshot of voting rights is generated. The current options are with `block.timesamp` or `block.number`, which only have block-level granularity. But the world state changes across transactions within a block. If there is any transaction in the same block that causes a voting right to change, the block number or block timestamp is not sufficient to pinpoint the world state of a snapshot of voting rights.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+1. Parameter Table
+
+- `FORK_NUM_TBD`: the fork block number to start introducing this change.
+- `OPCODE_TBD`: the opcode value being introduced.
+
+1. Upon `TBD_FORK_NUM`, add EVM an opcode with name `TXNUM` and value `TBD_OPCODE` instruction to retrieve TX number.
```

Comment:
```suggestion
1. Upon `TBD_FORK_NUM`, add EVM an opcode with name `TXNUM` and value `TBD_OPCODE` instruction to retrieve TX number.
```
This sentence is confusing.

Reply:
I updated the wording, can you see if it improves?